src/components/ToDoComponent/TodoComponent.tsx

```diff
@@ -0,0 +1,47 @@
+import { Text, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import Card from './Card';
+import data from './Data';
+import { TodoStyles } from './Styles/TodoStyles';
+
+const TodoComponent = () => {
+  const [tasks, setTasks] = useState(data);
+  const [position, setpostion] = useState(1);
+
+  const changeCardFunction = (id: any) => {
+    const newData = tasks.filter((item) => item.id == id);
+    tasks.shift();
+    tasks.push(newData[0]);
+    position !== tasks[tasks.length - 1].id
+      ? setpostion(tasks[position + 1].id)
+      : setpostion(tasks[0].id);
+  };
+
+  const removeCard = (id: any) => {
+    const newdata = data.filter((item) => item.id !== id);
+    setTasks(newdata);
+  };
+  return (
+    <View style={TodoStyles.container}>
+      <Text style={TodoStyles.title}>To Do's</Text>
+      <View style={{ paddingVertical: 35 }}>
+        {tasks
+          .map((task) => {
```

Comment:
we can use flatlist here

Reply:
Reason why I did not added flatlist here is that I want to make only one component visible in the beginning which is the animated one and, and the others have a position absolute, so everytime the state changes I want to again map over the array and change the styling and make the card which is in the next index the current card which will be the animated card.